raiden/network/transport/matrix.py

```diff
@@ -662,58 +737,58 @@ def _get_room_for_address(
         )
         return room
 
-    def _get_unlisted_room(self, room_name, invitees):
+    def _get_unlisted_room(self, room_name, invitees: List[User]):
         """Obtain a room that cannot be found by search_room_directory."""
         room_name_full = f'#{room_name}:{self._server_name}'
-        room_not_found = False
+        invitees_uids = [user.user_id for user in invitees]
 
-        for _ in range(10):
-            if room_not_found:
-                try:
-                    room = self._client.create_room(
-                        room_name,
-                        invitees=[user.user_id for user in invitees],
-                        is_public=True,  # FIXME: debug only
+        for _ in range(5):
```

Comment:
These values are arbitrary, the loop should never happen more than 2x (or it can join, or it fails the creation because of the race condition of another node creating the room at the same time, and then it joins the room on next loop), these `5` to `10` values are just for _safety_, and I put a smaller one here because we allow unnamed rooms as a safe measure in a case where the last user leaves a room and it can't be re-joined nor created with the canonical name after.